lib/puppet/provider/package/dpkg.rb

```diff
@@ -113,6 +120,20 @@ def query
 
     # list out our specific package
     begin
+      if @resource.allow_virtual?
```

Comment:
For rpm/yum, we first query for the exact match and fallback to trying the virtual package. Should dpkg/apt do the same?